pkg/storage/secret/metadata/keeper_model.go

```diff
@@ -246,3 +246,53 @@ func toProvider(keeperType secretv0alpha1.KeeperType, payload string) secretv0al
 		return nil
 	}
 }
+
+// extractSecureValues extracts unique securevalues referenced by the keeper, if any.
+func extractSecureValues(kp *secretv0alpha1.Keeper) map[string]struct{} {
+	secureValuesFromAWS := func(aws secretv0alpha1.AWSCredentials) map[string]struct{} {
```

Comment:
I think I made those functions to be re-used for the SQL Keeper, but since we won't have them anymore, we could simplify a bit the code. I will do that in the feature branch and we can merge it as-is to keep moving!